src/main/java/com/gruzini/tennistico/mappers/MatchInfoParser.java

```diff
@@ -0,0 +1,70 @@
+package com.gruzini.tennistico.mappers;
+
+import com.gruzini.tennistico.domain.Match;
+import com.gruzini.tennistico.domain.Player;
+import com.gruzini.tennistico.domain.Score;
+import com.gruzini.tennistico.exceptions.PlayerNotFoundException;
+import com.gruzini.tennistico.services.entity_related.MatchService;
+import com.gruzini.tennistico.services.entity_related.ScoreService;
+import org.springframework.stereotype.Component;
+
+import static java.util.Objects.isNull;
+
+@Component
+public class MatchInfoParser {
+    private final MatchService matchService;
+    private final ScoreService scoreService;
+
+    public MatchInfoParser(MatchService matchService, ScoreService scoreService) {
+        this.matchService = matchService;
+        this.scoreService = scoreService;
+    }
+
+    public String getScore(final Long matchId, final Player player) {
+        if (isNull(matchId)){
+            return "";
+        }
+        final Match match = matchService.getById(matchId);
+        if (isNull(match.getScore())) {
+            return "";
+        }
+        return getDivertedScoreString(match, player);
+    }
+
+    private String getDivertedScoreString(final Match match, final Player player) {
+        StringBuilder stringBuilder = new StringBuilder();
+        final Score score = scoreService.getScoreWithSets(match.getScore().getId());
+        score.getSets().forEach(set -> {
+            if (player.equals(match.getHost())) {
+                appendSet(stringBuilder, set.getHostScore(), set.getGuestScore());
+            } else {
+                appendSet(stringBuilder, set.getGuestScore(), set.getHostScore());
+            }
+        });
+        return stringBuilder.toString();
+    }
+
+    private void appendSet(StringBuilder stringBuilder, int recipientScore, int opponentScore) {
```

Comment:
Osobiście wydaje mi się, że wyciągnięcie takiej prywatnej metody jest trochę bez sensu, tym bardziej, że powstaje jakiś dziwny void potworek, z trzema iniektowanymi zmiennymi.

Reply:
Po kilku naradach, wyszedł pomysł, że można by było zrobić to mniej więcej w ten sposób:
`
private String getDivertedScoreString(final Match match, final Player player) {
        StringBuilder stringBuilder = new StringBuilder();
        final Score score = scoreService.getScoreWithSets(match.getScore().getId());
        score.getSets().forEach(set -> {
            if (player.equals(match.getHost())) {
                appendSet(set.getHostScore(), set.getGuestScore());
            } else {
                appendSet(set.getGuestScore(), set.getHostScore());
            }
        });
        return stringBuilder.toString();
    }
    private String appendSet(int recipientScore, int opponentScore) {
        return recipientScore + "-" + opponentScore + " ";
    }
`
coby nie przekazywać StringBuildera i żeby metoda nie była void.

Reply:
no ok

Reply:
Btw, w pierwszej metodzie możnaby returnować stringBuilder.toString().trim() coby pozbyć się tej luźnej spacji na końcu. Tak mi przyszło do głowy przy okazji.